File: Army(java)/Warriors/Rogue.java
package Warriors;

import EnumID.*;
import Basic.*;

public class Rogue extends Unit {
    private WarriorID warrior;
    
    public Rogue(String name, int hp, int dmg) {
        super(name, hp, dmg);
        this.warrior = WarriorID.ROGUE;
    }

    public WarriorID getWarriorID() {
        return warrior;
    }

    public void setWarriorID(WarriorID newID) {
        warrior = newID;
    }
}

File: Army(c++)/Warriors/Rogue.cpp
#include "Rogue.hpp"

Rogue::Rogue(const std::string& name, int hp, int dmg) 
    : Unit(name, hp, dmg, new SoldierState(this))
    , warrior(WarriorID::ROGUE) {}

Rogue::~Rogue() {}

WarriorID Rogue::getWarriorID() const {
    return this->warrior;
}

void Rogue::setWarriorID(WarriorID newID) {
    this->warrior = newID;
}
